### PR TITLE
docs: update remote API responses and minor fixes

### PR DESCRIPTION
**Pull request by thaJeztah**
Add back the "old" networksettings fields that were removed,
but added back to maintain backward compatibility, in
docker#17538

Update network endpoint responses, with updated response
introduced in;
docker#17536

Added changes to v1.22 that were applied to the v1.21 / v1.20 docs
after the API bump(s);

docker#17085
docker#17127
docker#13707

Also fixed some mixed tab/spaces indentation
and Markdown formatting issues (causing code-blocks to
be rendered incorrectly)

### CODE REVIEW

File: docs/reference/api/docker_remote_api_v1.21.md [hunk 1]
```diff
@@ -441,24 +441,32 @@ Return low-level information on the container `id`
...
+			"IPAddress": "",
+			"IPPrefixLen": 0,
+			"IPv6Gateway": "",
+			"MacAddress": "",
 			"Networks": {
 				"bridge": {
 					"EndpointID": "",
 					"Gateway": "",
-					"IPAdress": "",
+					"IPAddress": "",
```
Comment (thaJeztah) [inline docs/reference/api/docker_remote_api_v1.21.md]: @tiborvass this was the one that surprised me

Comment (tiborvass) [inline docs/reference/api/docker_remote_api_v1.21.md]: agreed

**Comment by mavenugo** [comment]
@thaJeztah @tiborvass i have some important updates to the docs as well. Shall I use this PR to update them ? Atleast it will help avoid multiple PRs to track and cherry-pick.

**Comment by thaJeztah** [comment]
@mavenugo are those the network case-change docs? (i.e. capitalize in the JSON response)?

**Comment by tiborvass** [comment]
@mavenugo either way is fine for me.

**Comment by thaJeztah** [comment]
@tiborvass just chatted up with @mavenugo , and those changes are not API related, so he'll create a separate PR, thanks!

File: docs/reference/api/docker_remote_api_v1.21.md [hunk 2]
```diff
@@ -2681,10 +2738,10 @@ Status Codes:
...
 JSON Parameters:
 
-- **name** - The new network's name. this is a mandatory field
-- **driver** - Name of the network driver to use. Defaults to `bridge` driver
-- **options** - Network specific options to be used by the drivers
-- **check_duplicate** - Requests daemon to check for networks with same name
+- **Name** - The new network's name. this is a mandatory field
+- **Driver** - Name of the network driver to use. Defaults to `bridge` driver
+- **Options** - Network specific options to be used by the drivers
+- **CheckDuplicate** - Requests daemon to check for networks with same name
```
Comment (thaJeztah) [inline docs/reference/api/docker_remote_api_v1.21.md]: I noticed `IPAM` was missing here, but didn't know how that should be used; https://github.com/docker/docker/pull/17536/files#diff-8bd32afa592e764a30005657f5494613R360

File: docs/reference/api/docker_remote_api_v1.21.md [hunk 3]
```diff
@@ -2603,39 +2655,44 @@ Status Codes:
...
+  "Containers": {
+    "39b69226f9d79f5634485fb236a23b2fe4e96a0a94128390a7fbbcc167065867": {
+      "EndpointID": "ed2419a97c1d9954d05b46e462e7002ea552f216e9b136b80a7db8d98b442eda",
+      "MacAddress": "02:42:ac:11:00:02",
+      "IPv4Address": "172.17.0.2/16",
+      "IPv6Address": ""
     }
+  },
+  "Options": {
+    "com.docker.network.bridge.default_bridge": "true",
```
Comment (thaJeztah) [inline docs/reference/api/docker_remote_api_v1.21.md]: these options appeared in the output, but wasn't sure these should be included in the documentation

@mavenugo ?

File: docs/reference/api/docker_remote_api_v1.21.md (hunk 2 above)
Comment (thaJeztah) [inline docs/reference/api/docker_remote_api_v1.21.md]: oh @mavenugo ^^

**Comment by thaJeztah** [comment]
updated with `/network` changes

@moxiegirl @mavenugo PTAL

File: docs/reference/api/docker_remote_api_v1.21.md (hunk 2 above)
Comment (mavenugo) [inline docs/reference/api/docker_remote_api_v1.21.md]: @thaJeztah good point. we might be missing information on IPAM Driver in the docs. I will work with @moxiegirl on that.

File: docs/reference/api/docker_remote_api_v1.21.md (hunk 3 above)
Comment (mavenugo) [inline docs/reference/api/docker_remote_api_v1.21.md]: Its okay to be included. it will be seen by default for the `default` bridge network.

**Comment by moxiegirl** [comment]
LGTM

**Comment by thaJeztah** [comment]
@mavenugo can you give this a "lgtm" if it lgty? Just to double check :+1:

**Comment by mavenugo** [comment]
@thaJeztah do you want us to fix the IPAM in this PR or do another 1 ?

**Comment by thaJeztah** [comment]
@mavenugo another one should be fine; let's take it one step at a time?

**Comment by mavenugo** [comment]
thanks @thaJeztah LGTM

**Comment by thaJeztah** [comment]
Thanks for looking Madhu

**Comment by thaJeztah** [comment]
@tiborvass please cherry-pick

**Pull request merged**